mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -47,12 +46,17 @@ public CassandraAnnotationMapper(CassandraId mailboxId, Session session) {
         this.session = session;
     }
 
-    public List<MailboxAnnotation> getAnnotations() {
-        return getResult(getStoredAnnotationRow());
+    public List<MailboxAnnotation> getAllAnnotations() {
+        return getStoredAnnotationRow()
+                .map(this::toAnnotation)
+                .collect(Collectors.toList());
     }
 
-    public List<MailboxAnnotation> getAnnotations(List<String> keys) {
-        return getResult(getStoredAnnotationRow(keys));
+    public List<MailboxAnnotation> getAnnotationsByKeys(List<String> keys) {
+        return getStoredAnnotationRow()
+                .filter(row -> keys.contains(row.getString(CassandraAnnotationTable.KEY)))
```

Comment:
Yes, I misundertood your meaning when you said "filter in the method". It's changed as your new comment